File: solutions/dart/bob/1/lib/bob.dart
// Dedicated to Shree DR.MDD – the guiding force behind every byte I write.

RegExp caps = RegExp(r'\p{Uppercase}', unicode: true);
RegExp smalls = RegExp(r'\p{Lowercase}', unicode: true);

class Bob {
  String response(String input) {
    final phrase = input.trimRight();

    final muted = phrase.isEmpty;
    final loud = phrase.contains(caps) && !phrase.contains(smalls);
    final interrogative = phrase.endsWith('?');

    return switch ((muted, loud, interrogative)) {
      (true, _, _) => 'Fine. Be that way!',
      (_, true, true) => "Calm down, I know what I'm doing!",
      (_, true, false) => 'Whoa, chill out!',
      (_, false, true) => 'Sure.',
      _ => 'Whatever.',
    };
  }
}

File: solutions/nim/bob/1/bob.nim
# Dedicated to Shree DR.MDD
import strutils

proc isYelling(msg: string): bool =
  msg.toUpperAscii == msg and msg.contains(Letters)

proc isQuery(msg: string): bool =
  msg.strip.endsWith('?')

proc isMute(msg: string): bool =
  msg.isEmptyOrWhitespace

proc hey*(msg: string): string =
  if msg.isMute:
    "Fine. Be that way!"
  elif msg.isYelling and msg.isQuery:
    "Calm down, I know what I'm doing!"
  elif msg.isYelling:
    "Whoa, chill out!"
  elif msg.isQuery:
    "Sure."
  else:
    "Whatever."
